src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpTypeStyleDiagnosticAnalyzerBase.cs

```diff
@@ -11,12 +11,84 @@
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Text;
+using static Microsoft.CodeAnalysis.CSharp.Diagnostics.TypeStyle.CSharpTypeStyleDiagnosticAnalyzerBase;
 
 namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.TypeStyle
 {
+    internal abstract partial class CSharpTypeStyleHelper
+    {
+        internal abstract bool IsStylePreferred(SemanticModel semanticModel, OptionSet optionSet, State state, CancellationToken cancellationToken);
+        internal abstract bool TryAnalyzeVariableDeclaration(TypeSyntax typeName, SemanticModel semanticModel, OptionSet optionSet, CancellationToken cancellationToken);
+        protected abstract bool AssignmentSupportsStylePreference(SyntaxToken identifier, TypeSyntax typeName, ExpressionSyntax initializer, SemanticModel semanticModel, OptionSet optionSet, CancellationToken cancellationToken);
+
+        internal TypeSyntax CanOfferFix(SyntaxNode declarationStatement, SemanticModel semanticModel, CancellationToken cancellationToken)
```

Comment:
argument order tends to go from broadest to narrowest.  So SemanticModel would be first, then the node.

Reply:
I'm matching the order of other methods in this file. I don't want to create churn for such refactoring. I'll keep as-is for now if ok.

---
In reply to: [172284262](https://github.com/dotnet/roslyn/pull/25208#discussion_r172284262) [](ancestors = 172284262)